frontend/src/views/Lobby.tsx

```diff
@@ -330,88 +396,116 @@ function LobbyPage() {
-
-          // Set duration to undefined to allow users to clear field
-          if (!value) {
-            setDuration(undefined);
-          } else {
-            const newDuration = Number(value);
-            if (newDuration >= 0 && newDuration <= 60) {
-              setDuration(newDuration);
+            codejoust.co/play?room=
+            {currentRoomId}
+            <InlineCopyIcon>content_copy</InlineCopyIcon>
+          </InlineBackgroundCopyText>
+          {' '}
+          or at
+          {' '}
+          <i>codejoust.co/play</i>
+          {' '}
+          with Room ID:
+        </MainHeaderText>
+        <IdContainer id={currentRoomId} />
+        <PrimaryButtonZeroLeftMargin
+          onClick={handleStartGame}
+          disabled={loading || !isHost(currentUser)}
+          title={!isHost(currentUser) ? 'Only the host can start the game' : undefined}
+        >
+          Start Game
+        </PrimaryButtonZeroLeftMargin>
+      </HeaderContainer>
+
+      <FlexBareContainerLeft>
+        <PlayersContainer>
+          <SmallHeaderTextZeroTopMargin>
+            Players
+            {
+              (activeUsers && inactiveUsers)
+                ? ` (${activeUsers.length + inactiveUsers.length})`
+                : null
             }
-          }
-        }}
-        onBlur={updateRoomDuration}
-      />
-      <br />
-
-      {isHost(currentUser)
-        ? (
-          <PrimaryButton
-            color={ThemeConfig.colors.gradients.blue}
-            onClick={handleStartGame}
-            disabled={loading}
-          >
-            Start Game
-          </PrimaryButton>
-        )
-        : <MediumText>Waiting for the host to start the game...</MediumText>}
-    </div>
+          </SmallHeaderTextZeroTopMargin>
+          <BackgroundContainer>
+            {
+              displayUsers(activeUsers, true)
+            }
+            {
+              displayUsers(inactiveUsers, false)
+            }
+            { error ? <ErrorMessage message={error} /> : null }
+            { loading ? <Loading /> : null }
+          </BackgroundContainer>
+        </PlayersContainer>
+        <RoomSettingsContainer>
+          <SmallHeaderTextZeroTopMargin>Room Settings</SmallHeaderTextZeroTopMargin>
+          <BackgroundContainer>
+            <NoMarginMediumText>Difficulty</NoMarginMediumText>
+            <DifficultyContainer>
+              {Object.keys(Difficulty).map((key) => {
+                const difficultyKey: Difficulty = Difficulty[key as keyof typeof Difficulty];
+                return (
+                  <SmallDifficultyButton
+                    difficulty={difficultyKey}
+                    onClick={() => updateDifficultySetting(key)}
+                    active={difficulty === difficultyKey}
+                    enabled={isHost(currentUser)}
+                    disabled={!isHost(currentUser)}
+                    title={!isHost(currentUser) ? 'Only the host can change these settings' : undefined}
```

Comment:
Is there any way to make this + the start game button more obvious that it's host-only? The title usually doesn't show up until you leave your cursor unmoving for a couple seconds (at first I was gonna suggest adding a title but then just now saw you already had it lol)

Reply:
Yeah that's a good idea - I'll make some tooltip appear immediately on hover, as I agree that the "title" isn't a good solution here.

Reply:
Ok, I spent a very long time on a solution here, because I wanted the tooltip to move around with the mouse and it had to show only on disabled elements. Since disabled elements don't fire events, I then created a div parent wrapper around those elements, transferring the margin from the original elements to the container ones. Following that, I realized that this solution still didn't work, as this still prevented the divs from firing elements on the specific area where the disabled element is. The final solution has the wrapper div with the attribute `position: relative`, then I add a child element (sibling to the original disabled element) with absolute positioning that has the identical height and width of the disabled element, and a z-index of one greater.

This now works, though it's certainly a complicated solution. I shied away from simply removing the "disabled" component as I did want to keep that integrity, though I'd understand if that might be a better solution given this one's complexity. I think this should all work fine now, but just wanted to note down the process here 👍

Reply:
I think it looks great! thanks for doing that